pkg/producer/kafka.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package producer
+
+import (
+	"fmt"
+
+	"github.com/Shopify/sarama"
+	"github.com/golang/protobuf/proto"
+	"k8s.io/klog"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/producer/protobuf"
+)
+
+var (
+	KafkaConfigVersion sarama.KafkaVersion
+	// ProtoSchemaTypes lists all proto schema supported in Kafka producer.
+	ProtoSchemaTypes = map[string]string{
+		"FlowType1": "FlowType1",
+		"FlowType2": "FlowType2",
+	}
+)
+
+type KafkaProducer struct {
+	producer        sarama.AsyncProducer
+	topic           string
+	protoSchemaType string
+}
+
+// convertIPFIXMsgToFlowMsgs converts data records in IPFIX message to flow messages
+// in given proto schema.
+func convertIPFIXMsgToFlowMsgs(msg *entities.Message, protoSchemaType string) []*protobuf.FlowMessage {
+	set := msg.GetSet()
+	if set.GetSetType() == entities.Template {
+		return nil
+	}
+
+	var convertRecordFunc convertRecordToFlowType
+	switch protoSchemaType {
+	case "FlowType1":
+		convertRecordFunc = convertRecordToFlowType1
```

Comment:
Can we save convertRecordToFlowType1 to KafkaProducer struct?

Reply:
sure.